src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs

```diff
@@ -754,6 +747,34 @@ private Task<LockFile> ReadLockFile(string projectLockFilePath)
                     frameworkInfo.Dependencies.Select(x => x.LibraryRange.ToString())));
             }
 
+            // Record all libraries used
+            foreach (var item in graphItems.OrderBy(x => x.Match.Library, new LibraryComparer()))
+            {
+                var library = item.Match.Library;
+                var packageInfo = repository.FindPackagesById(library.Name)
+                    .FirstOrDefault(p => p.Version == library.Version);
+
+                if (packageInfo == null)
+                {
+                    continue;
+                }
+
+                var package = packageInfo.Package;
+
+                LockFileLibrary previousLibrary = null;
+                previousLibraries?.TryGetValue(Tuple.Create(library.Name, library.Version), out previousLibrary);
+
```

Comment:
Could you add an assert here saying `previousLibrary` is not expected to be null? Or Does `CreateLockFileLibrary` handle nulls?